programs/ssl/ssl_client2.c

```diff
@@ -2680,7 +2681,7 @@ int main( int argc, char *argv[] )
         {
             mbedtls_printf( " failed\n  ! mbedtls_ssl_set_cid returned %d\n\n",
                             ret );
-            return( ret );
+            goto exit;
```

Comment:
It would be better not to mix this change with the changes related to `query_config` in the same commit. Those are two different things and then in the following commit it is not obvious if the changes are related to this change or to the `query_config` change. Do first the fixes/changes only related to `query_config` in one or several commits, then fix that one, again in one or several commits.